scripts/python/update_snippet/python_snippet_updater.py

```diff
@@ -0,0 +1,102 @@
+import sys
+import logging
+from pathlib import Path
+import argparse
+import re
+from typing import Dict
+
+_LOGGER = logging.getLogger(__name__)
+
+snippets = {}
+not_up_to_date = False
+
+target_snippet_sources = ["samples/*.py", "samples/*/*.py"]
+target_md_files = ["README.md"]
+
+def check_snippets() -> Dict:
+    return snippets
+
+def check_not_up_to_date() -> bool:
+    return not_up_to_date
+
+def get_snippet(file: str) -> None:
+    file_obj = Path(file)
+    with open(file_obj, 'r') as f:
+        content = f.read()
+    pattern = "# \\[START(?P<name>[A-Z a-z0-9_]+)\\](?P<body>[\\s\\S]+?)# \\[END[A-Z a-z0-9_]+\\]"
+    matches = re.findall(pattern, content)
+    for match in matches:
+        s = match
+        name = s[0].strip()
+        snippet = s[1]
+        # Remove extra spaces
+        spaces = ""
```

Comment:
I'd really appreciate a little code comment or something that maps out what is actually happening here. I am not convinced of this code as-is just from reading it.

A snippet is the complete code from the actual python file. So ripped directly from the python file it'll look like.

```
        # [START trio]
        from azure.core.pipeline.transport import TrioRequestsTransport

        async with AsyncPipeline(TrioRequestsTransport(), policies=policies) as pipeline:
            return await pipeline.run(request)
        # [END trio]
```

And we need to _remove_ spaces, as we will be emplacing the snippet into the markdown file.

First question, why do we start with the second character of the snippet? Shouldn't we start with the first?

```
        if snippet[-1] == "\n":
            snippet = snippet[:-1]
```

I seriously do not get it.